agent_application/src/main.rs

```diff
@@ -42,22 +42,40 @@ async fn main() -> io::Result<()> {
 
     let (identity_state, issuance_state, holder_state, verification_state) = match config().event_store.type_ {
         EventStoreType::Postgres => (
-            agent_store::identity_state::<Postgres>(identity_services, identity_event_publishers).await,
+            agent_store::identity_state::<Postgres>(&Postgres, identity_services, identity_event_publishers).await,
```

Comment:
nit: Now all the `_state` functions expect `builder: &CCB` the type checker doesn't need the `::<Postgres/MongoDb/InMemory>` 'hints' anymore so you could omit those to make the formatting of this match block a bit leaner.